tests/utils.py

```diff
@@ -7,6 +7,23 @@
 from typing import Optional
 
 
+@contextmanager
```

Comment:
Given that the flow will be run in a subprocess, maybe better to patch the subprocess env (i.e. add `run_flow(env_overrides={'FOO':'bar'})` and create an `env` copy for `subprocess`) than to overwrite the main environment

Reply:
The subprocess env inherits the main environment anyway, and the context manager reverts changes so there shouldn't be any incorrectness?

This approach benefits from not having to pass arguments down (if future functions have deeper nesting), and works for any tests which don't run in `subprocess` but do use env vars.

Reply:
I guess I was just thinking that since `subprocess` provides this exact functionality that it was unnecessary to implement it ourselves, but actually with the suggested change (added just now) it actually brings the amount of code to a pretty minimal level